cli/azd/cmd/mcp.go

```diff
@@ -6,91 +6,688 @@ package cmd
 import (
 	"context"
 	"fmt"
+	"io"
+	"strings"
 
 	"github.com/azure/azure-dev/cli/azd/cmd/actions"
 	"github.com/azure/azure-dev/cli/azd/internal"
-	"github.com/azure/azure-dev/cli/azd/pkg/alpha"
-	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/internal/agent/consent"
+	"github.com/azure/azure-dev/cli/azd/internal/mcp/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
 	"github.com/azure/azure-dev/cli/azd/pkg/input"
-	"github.com/azure/azure-dev/cli/azd/pkg/llm/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/output/ux"
 	"github.com/mark3labs/mcp-go/server"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 )
 
-func newMcpFlags(cmd *cobra.Command, global *internal.GlobalCommandOptions) *mcpFlags {
-	flags := &mcpFlags{}
-	flags.Bind(cmd.Flags(), global)
+// Register MCP commands
+func mcpActions(root *actions.ActionDescriptor) *actions.ActionDescriptor {
+	group := root.Add("mcp", &actions.ActionDescriptorOptions{
+		Command: &cobra.Command{
+			Use:   "mcp",
+			Short: "Manage Model Context Protocol (MCP) server.",
+		},
+		GroupingOptions: actions.CommandGroupOptions{
+			RootLevelHelp: actions.CmdGroupAlpha,
+		},
+	})
 
-	return flags
-}
+	// azd mcp start
+	group.Add("start", &actions.ActionDescriptorOptions{
+		Command: &cobra.Command{
+			Use:   "start",
+			Short: "Starts the MCP server.",
+			Long: `Starts the Model Context Protocol (MCP) server.
 
-func newMcpCmd() *cobra.Command {
-	return &cobra.Command{
-		Use:   "mcp",
-		Short: "Run MCP server.",
-	}
+This command starts an MCP server that can be used by MCP clients to access
+azd functionality through the Model Context Protocol interface.`,
+			Args: cobra.NoArgs,
+		},
+		OutputFormats:  []output.Format{output.NoneFormat},
+		DefaultFormat:  output.NoneFormat,
+		ActionResolver: newMcpStartAction,
+		FlagsResolver:  newMcpStartFlags,
+	})
+
+	// azd mcp consent subcommands
+	consentGroup := group.Add("consent", &actions.ActionDescriptorOptions{
+		Command: &cobra.Command{
+			Use:   "consent",
```

Comment:
I'd like to talk more about this consent feature...

Should azd ask for consent when running built-in tools? 

It would be 100% understandable if a user can add external mcp servers to azd. Then azd would require user confirmation before calling those external tools. But it does not make sense to me when azd is asking consent to run something that is within azd's implementation.

For example, azd does not ask user for consent to run commands like `npm build` or `docker build`. Not even to run customer's hooks.

@tg-msft, what's your thoughts here?

Reply:
I will skip looking to all the consent command and components for now.

I think all of that should be moved to another PR

Reply:
A related question I have is how individual azd commands fit into tools. For example in the project validation stage, when we instruct it to run `azd provision --preview`, is there a specific tool for that?

And do we differentiate between "read-only" commands like `azd provision --preview` and ones with "write" capabilities like `azd provision` or `azd deploy`? One concern is if it ends up running something potentially destructive or expensive.

Reply:
Right now it invokes azd command or any other commands in a terminal type execute.  For example VS code always asks users to consent to executing commands in a terminal.

100% get your point about our "build-in" tools.  Happy to have that discussion but we can change all of this.  No harm in just getting some initial version out there and then just figuring these things out.